tests/appsec/waf/test_addresses.py

```diff
@@ -150,14 +152,16 @@ def test_cookies(self):
     @bug(library="ruby")
     def test_cookies_with_special_chars(self):
         """Other cookies patterns, to be merged once issue are corrected"""
-        r = self.weblog_get("/waf", cookies={"value": ";shutdown--"})
+        # weblog_get doesn't properly escape semicolons in the cookie value,
```

Comment:
Is it specific for java ?

Reply:
I'm doing PHP, not Java, but no, it's not specific. If the cookie header has a semicolon, the http servers will interpret that as starting a new cookie, not as part of the value of the cookie. Arguably, the method `weblog_get` should be changed instead so that int correct encodes the value.